crates/isograph_schema/src/magic_mutation_fields.rs

```diff
@@ -496,3 +435,264 @@ fn skip_arguments_contained_in_field_map(
+        bytes byte_buf option unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum ignored_any identifier
+    }
+}
+
+impl<'de, TName> Deserializer<'de> for ValueSerializer<'de, TName, ConstantValue> {
+    type Error = ProcessTypeDefinitionError;
+
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: de::Visitor<'de>,
+    {
+        let deserializer = ConstantValueDeserializer {
+            value: &self.name_value_pair.value.item,
+        };
+        deserializer.deserialize_any(visitor)
+    }
+
+    serde::forward_to_deserialize_any! {
+        bool i8 i16 i32 i64 i128 u8 u16 u32 u64 u128 f32 f64 char str string
+        bytes byte_buf option unit unit_struct newtype_struct seq tuple
+        tuple_struct map struct enum ignored_any identifier
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    use common_lang_types::TextSource;
+    use graphql_lang_types::{
+        ConstantValue, GraphQLDirective, GraphQLTypeSystemExtension,
+        GraphQLTypeSystemExtensionOrDefinition,
+    };
+    use graphql_schema_parser::*;
+    use intern::string_key::Intern;
+    use std::error::Error;
+
+    fn unwrap_directive(
+        extension_or_definition: GraphQLTypeSystemExtensionOrDefinition,
+    ) -> Result<Vec<GraphQLDirective<ConstantValue>>, Box<dyn Error>> {
+        if let GraphQLTypeSystemExtensionOrDefinition::Extension(extension) =
+            extension_or_definition
+        {
+            let GraphQLTypeSystemExtension::ObjectTypeExtension(object_type_extension) = extension;
+            return Ok(object_type_extension.directives.clone());
+        }
+        Err("unexpected structure of directive".into())
+    }
+
+    #[test]
+    fn test_magic_mutation_deserialization() -> Result<(), Box<dyn Error>> {
+        let source = "extend type Mutation
+        @exposeField(
+          field: \"set_pet_tagline\"
+          path: \"pet\"
+          field_map: [{ from: \"id\", to: \"input.id\" }]
+        )
+        @exposeField(
+          field: \"set_pet_best_friend\"
+          path: \"pet\"
+          field_map: [{ from: \"id\", to: \"id\" }]
+        )";
+        let text_source = TextSource {
+            path: "dummy".intern().into(),
+            span: None,
+        };
+        let document = parse_schema_extensions(source, text_source).map_err(|e| e.item)?;
+        let directives = document
+            .0
+            .into_iter()
+            .map(|dir| unwrap_directive(dir.item))
+            .collect::<Result<Vec<_>, _>>()?;
+        let directives: Vec<GraphQLDirective<ConstantValue>> =
+            directives.into_iter().flatten().collect();
+
+        let magic_mutations: Result<Vec<MagicMutationFieldInfo>, _> = directives
+            .into_iter()
+            .map(|directive| from_graph_ql_directive::<MagicMutationFieldInfo>(&directive))
+            .collect();
```

Comment:
let's not test multiple directives in the same unit test

Reply:
What a rookie mistake :( will fix